tools/generate_ensembl_json.py

```diff
@@ -0,0 +1,475 @@
+#!/usr/bin/env python
+"""
+Retrieve gene/transcript/exon/domain information from Ensembl and build a set
+of annotations in custom json format for use with MAVIS. Uses 'pyensembl'
+module to retrieve gene, transcript, and exon information from the Ensembl
+REST-API. The pyensembl module does not have a method for retrieving domain
+information, so manually retrieve domain information for each transcript from
+the API, and cache it along with the cache files from pyensembl.
+"""
+import argparse
+import csv
+import os
+import sys
+import time
+from collections import defaultdict
+from functools import wraps
+from glob import glob
+
+import requests
+import simplejson as json
+from pyensembl import EnsemblRelease
+
+
+VERSION = "1.0.0"
+SCRIPT = os.path.abspath(__file__)
+CACHE_DEFAULT = os.environ["HOME"] + "/.cache"
+
+DOMAIN_CACHE = {}
+DOMAIN_CACHE_PATH = None
+
+
+def get_date(date_str=None):
+    """
+    Returns the date (str) as <weekday> <month> <day> <time hh:mm:ss> <year>
+    """
+    if date_str:
+        return time.asctime(time.localtime(date_str))
+    else:
+        return time.asctime(time.localtime(time.time()))
+
+
+def parse_best_file(path):
```

Comment:
we haven't been using the best transcript files anymore, just going off the 'canonical' transcript listed by ensembl